orca-retrofit2/src/main/kotlin/com/netflix/spinnaker/orca/retrofit2/NetworkingException.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.retrofit2
+
+import okhttp3.Request
+import java.io.IOException
+
+class NetworkingException(
+  val request: Request,
+  cause: IOException
+) : RuntimeException(cause)
```

Comment:
Do we want this to extend `HttpException` instead? Left as-is, we'll regress in error handling behavior since `RetrofitError` encompassed both HTTP- & network-level errors.

Reply:
I was wondering about that but I didn't want to no-op things like `code()` on the sub-class. Possibly we need an overall exception that can contain a cause that is either `HttpException` or `IOException`.

Reply:
I like that. I just don't want us to have to convert from a single `catch (RetrofitError)` to now catching two different exceptions, I feel a refactor like that will have a bunch of unintended consequences.

Reply:
The more I think about it, the more I suspect the converter that Adam linked is probably the right way forward.

Reply:
You mean the exception handler?

Reply:
Yes.